src/html5_parser/parser/document.rs

```diff
@@ -50,6 +85,29 @@ impl Document {
         }
     }
 
+    pub fn shared() -> DocumentHandle {
+        DocumentHandle(Rc::new(RefCell::new(Self::new())))
+    }
```

Comment:
The name `Document::shared()` is a little weird.  But the interface can be dialed in in a later PR.